File: source/includes/fts-tutorial/compound-constant-query.java
import java.util.Arrays;
import static com.mongodb.client.model.Filters.eq;
import static com.mongodb.client.model.Aggregates.limit;
import static com.mongodb.client.model.Aggregates.project;
import static com.mongodb.client.model.Projections.computed;
import static com.mongodb.client.model.Projections.excludeId;
import static com.mongodb.client.model.Projections.fields;
import static com.mongodb.client.model.Projections.include;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;

public class CompoundConstantQuery {
  public static void main( String[] args ) {
    // define query
    Document agg = new Document("filter", Arrays.asList(new Document("range", 
      new Document("path", "year").append("gte", 2013).append("lte", 2015)))).append("should", Arrays.asList(
      new Document("text", new Document("query", "snow").append("path", "title").append("score", 
        new Document("constant", new Document("value", 5))))));
	// specify connection
    String uri = "<connection-string>";
    // establish connection and set namespace
    try (MongoClient mongoClient = MongoClients.create(uri)) {
      MongoDatabase database = mongoClient.getDatabase("sample_mflix");
      MongoCollection<Document> collection = database.getCollection("movies");
	  // run query and print results		
      collection.aggregate(Arrays.asList(
        eq("$search", eq("compound", agg)), 
        limit(10), 
        project(fields(excludeId(), include("title", "year"), computed("score", new Document("$meta", "searchScore")))))
      ).forEach(doc -> System.out.println(doc.toJson()));	
    }
  }
}

File: source/includes/fts-tutorial/custom-score/compound-constant-query.js
const MongoClient = require("mongodb").MongoClient;
const assert = require("assert");

const agg = [
  {
    '$search': {
      'index': 'compound-query-custom-score-tutorial',
      'compound': {
        'filter': [
          {
            'range': {
              'path': 'year',
              'gte': 2013,
              'lte': 2015
            }
          }
        ],
        'should': [
          {
            'text': {
              'query': 'snow',
              'path': 'title',
              'score': {
                'constant': {
                  'value': 5
                }
              }
            }
          }
        ]
      },
      'highlight': {
        'path': 'title'
      }
    }
  }, {
    '$limit': 10
  }, {
    '$project': {
      '_id': 0,
      'title': 1,
      'year': 1,
      'score': {
        '$meta': 'searchScore'
      },
      'highlights': {
        '$meta': 'searchHighlights'
      }
    }
  }
];

MongoClient.connect(
  "<connection-string>",
  { useNewUrlParser: true, useUnifiedTopology: true },
  async function (connectErr, client) {
    assert.equal(null, connectErr);
    const coll = client.db("sample_mflix").collection("movies");
    let cursor = await coll.aggregate(agg);
    await cursor.forEach((doc) => console.log(doc));
    client.close();
  }
);
